### PR TITLE
Android debugger unixsockets

### PR DESCRIPTION
**Pull request by blagoev**
This pull request is releated to the changes here #273. A help is needed to merge this into master and then investigate what issues there might be in app builder CLI. 

The long story short. The android debugger now uses named sockets instead of ports hence the code rewrite in adndroid-debug-service to accommodate for that.
Contact me directly in order to merge this successfully into the respected main branches.

### CODE REVIEW

File: lib/services/android-debug-service.ts [hunk 1]
```diff
@@ -43,14 +45,64 @@ class AndroidDebugService implements IDebugService {
...
+            var server = net.createServer();
+            server.listen(candidatetPort, function (err: any) {
+                server.once('close', function () {
+                    future.return(true);
+                })
+                server.close();
+            });
+            
+            server.on('error', function (err: any) {
+               future.return(false);
```
Comment (Fatme) [inline lib/services/android-debug-service.ts]: maybe `future.throw`?

File: lib/services/android-debug-service.ts [hunk 2]
```diff
@@ -43,14 +45,64 @@ class AndroidDebugService implements IDebugService {
...
+            });
+            
+            return future;
+		}).future<boolean>()();
```
Comment (Fatme) [inline lib/services/android-debug-service.ts]: You don't need this line.

**Comment by blagoev** [comment]
@Fatme : Please fix all in place. This commit should be regarded as a reference for the necessary fixes that correspond to the related one in the engine.

**Comment by teobugslayer** [comment]
reimplemented in https://github.com/NativeScript/nativescript-cli/pull/1373

**Pull request closed without merging**